doc/api/process.markdown

```diff
@@ -186,6 +214,104 @@ this, you can either attach a dummy `.catch(() => { })` handler to
 `resource.loaded`, preventing the `'unhandledRejection'` event from being
 emitted, or you can use the [`'rejectionHandled'`][] event.
 
+## Event: 'warning'
+
+Emitted whenever Node.js emits a process warning. 
+
+A process warning is similar to errors in that they describe exceptional 
+conditions that are being brought to the user's attention. However, they are 
+not part of the normal Node.js and JavaScript error handling flow. 
+Node.js can emit warnings whenever it detects bad coding practices that could 
+lead to sub-optimal application performance, bugs or security vulnerabilities.
+
+The event handler for `'warning'` events is called with a single `warning` 
+argument whose value is an `Error` object. There are three key properties that 
+describe the warning:
+
+* `name` - The name of the warning (currently either `Warning`
+  or `BadPracticeWarning`).
+* `message` - A system-provided description of the warning.
+* `stack` - A stack trace to the location in the code where the warning was
+  issued.
+
+```js
+process.on('warning', (warning) => {
+  console.warn(warning.name);    // Print the warning name
+  console.warn(warning.message); // Print the warning message
+  console.warn(warning.stack);   // Print the stack trace
+});
+```
+
+By default, Node.js will print process warnings to `stderr`. The `--no-warnings`
+command-line option can be used to suppress the default console output but the
+`'warning'` event will still be emitted by the `process` object.
+
+The following example illustrates the warning that is printed to `stderr` when 
+two many listeners have been added to an event
+
+```
+$ node
+> event.defaultMaxListeners = 1;
+> process.on('foo', () => {});
+> process.on('foo', () => {});
+> (node:38638) BadPracticeWarning: Possible EventEmitter memory leak detected. 
+... 2 foo listeners added. Use emitter.setMaxListeners() to increase limit
+```
+
+In contrast, the following example turns off the default warning output and
+adds a custom handler to the `'warning'` event:
+
+```
+$ node --no-warnings
+> var p = process.on('warning', (warning) => console.warn('Do not do that!'));
+> event.defaultMaxListeners = 1;
+> process.on('foo', () => {});
+> process.on('foo', () => {});
+> Do not do that!
+```
+
+The `--trace-warnings` command-line option can be used to have the default
+console output for warnings include the full stack trace of the warning.
+
+### Warning Types
+
+Node.js currently produces two types of warnings:
+
+* `Warning` - a generic warning emitted by calling `process.emitWarning()`.
+* `BadPracticeWarning` - a warning indicating that parts of the Node.js API are
+  being used in a manner that is not recommended or has known issues.
+  
+### Emitting Custom Warnings
+
+The [`process.emitWarning(warning)`][process_emit_warning] method can 
+be used issue custom or application specific warnings.
```

Comment:
Yes, if the `name` of the warning is set to `DeprecationWarning`, then `'deprecation'` will be emitted rather than `'warning'`